### DISCUSSION TITLE
Receiving location in app after URL post

### DISCUSSION THREAD

**Discussion by ShaunBrassell** [HaylLtd/cordova-plugin-background-geolocation#103] [General]
Hi guys

I have the plugin working nicely in the background by calling the getLocations method when the app comes back into the foreground, and are then are able to draw the route taken on the map (thus not having to rely on flaky cordova JS calls in the background). 

I also need to post the locations to a server, which works well posting as the user goes along in the background, but these posted locations are then not retrievable via the getLocations call when the app returns to the foreground. I haven't dug through the Java code yet, but am guessing that once successfully posted, the location is marked for delete and thus not able to be pulled through via the getLocations?

**Comment by HarelM** [comment]
Yes, I believe this is a correct assumption...

**Reply by ShaunBrassell** [reply]
Thanks for the prompt reply as always Harel! Trying to go through the Java code to get a better understanding of the logic, but am a little stumped. It looks like getAllLocations  is programmed to return everything that is in the DB, no matter what the COLUMN_NAME_STATUS is. And getValidLocations will ommit all the ones with the deleted flag.

Then as far as I can see, entries in the DB are never actually deleted, just marked with a Deletion status tag (in COLUMN_NAME_STATUS)... only time they are actually removed is when you reach the maxThreshold. I looked at the deleteAllLocations code, it does look as its just updating the status codes in the DB, and leaving all entries there.

So far so good. When testing the code, if I have some locations in the DB, and I run getAllLocations, all are then shown. But if I then run deleteAllLocations, and redo getAllLocations, the return array is indeed empty ... which I did not expect.

Thus am thinking that getAllLocations method must be filtering out the ones with the delete tag, but I cant see that. So am missing something somewhere!

Anyway, will keep trying to understand whats going on, but please point me in the right direction if you know off hand. Have a great weekend!

**Reply by HarelM** [reply]
I think I had the same understanding when I started looking at the code, I don't remember to what conclusion I got at the end though.
It is over complicated unfortunately, but I don't have the right tests to simplify it :-(

**Reply by ShaunBrassell** [reply]
Glad its not just me! But at least the logic works then, am following your apps pattern, reading and deleting. Looks like I have also solved my issue of wanting to post to a server as well as view on the map ... commented out line 120 in the PostLocationTask class. Just need to check that I don't get any post duplicates, doesn't seem so in the few tests I have done :)
Will look at IOS at a later stage if this does work out.

if (mHasConnectivity && mConfig.hasValidUrl()) {
            if (postLocation(location)) {
                // mLocationDAO.deleteLocationById(locationId);      <--
                return; // if posted successfully do nothing more
            } else {
                mLocationDAO.updateLocationForSync(locationId);
            }
        } else {
            mLocationDAO.updateLocationForSync(locationId);
        }
